src/ORM/Query/DeleteQuery.php

```diff
@@ -0,0 +1,223 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.5.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Query;
+
+use Cake\Database\Connection;
+use Cake\Datasource\ResultSetInterface;
+use Cake\ORM\Query;
+use Cake\ORM\Table;
+
+/**
+ * Delete Query forward compatibility shim.
+ */
+class DeleteQuery extends Query
+{
+    /**
+     * Type of this query (select, insert, update, delete).
+     *
+     * @var string
+     */
+    protected $_type = 'delete';
+
+    /**
+     * Constructor
+     *
+     * @param \Cake\Database\Connection $connection The connection object
+     * @param \Cake\ORM\Table $table The table this query is starting on
+     */
+    public function __construct(Connection $connection, Table $table)
+    {
+        parent::__construct($connection, $table);
+        $this->from([$table->getAlias() => $table->getTable()]);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(?string $table = null)
+    {
+        $this->_deprecatedException('delete()', 'Remove this method call.');
```

Comment:
This isn't deprecated if we're throwing an exception. Somehow we need to communicate this call is not needed with the new DeleteQuery object.

Reply:
Should we be using regular deprecation warningsninstead of exceptions for along these methods then?

I was trying to make the required updates easy to find. However, using an exception will give a tougher upgrade than a warning would.

I will update these tongues a warning instead.